dask_jobqueue/pbs.py

```diff
@@ -60,66 +62,43 @@ class PBSCluster(JobQueueCluster):
 
     >>> cluster.adapt()
     """
+
+    #Override class variables
+    submit_command = 'qsub'
+    cancel_command = 'qdel'
+
     def __init__(self,
-                 name='dask',
-                 queue='regular',
+                 name='dask-worker',
+                 queue=None,
                  project=None,
-                 resource_spec='select=1:ncpus=36:mem=109GB',
-                 threads_per_worker=4,
-                 processes=9,
-                 memory='7GB',
+                 resource_spec='select=1:ncpus=24:mem=100GB',
```

Comment:
I'm not totally sure what to do here. One option is to leave this as `None` and force (most) users to set this. Keeping it as is may be okay too but everyone will have to change it (except for you :wink:). I have a slight preference for setting it to None.

Reply:
Yep I have the same concern. I found your defaults too high, so I set mines, but it is probably wrong too.

On our cluster if we don't configure it, the default will be `select=1:ncpus=1:mem=5GB`. So worker jobs will probably be killed promptly if dask worker is set with more than 1 thread or process. And I believe it would be really bad for PBS to launch one job for each 1 process dask-worker.

My preference would be more close to what is mentioned in #12, if nothing is provided, resource_spec is computed using dask-worker options. Still not enthusiast about that? It's a good compromise, keep resource_spec keyword, but build it from dask-worker options if not provided. The only concern if for memory limit, there may be a need to extrapolate from String to integer...

So maybe it is simpler to set it to None :)...

Reply:
We might inspect available memory and cpu cores using tools like psutil and multiprocessing

```python
In [1]: import psutil
In [2]: psutil.virtual_memory().total
Out[2]: 16680607744
In [3]: psutil.cpu_count()
Out[3]: 4
```

Reply:
That may be okay. The challenge is that you'll routinely have the situation where the machine that scheduler is running on is spec'd differently than the machines where the workers will be running on.  I think this may be worth doing as a first guess.

Reply:
I'm not inclined to base the resource_spec on the submitting node resource...
I would much prefer to guess it from the given threads and processes. Else I would default to None, maybe with a warning?

Reply:
These will tend to be different on every machine - what about incorporating a user config file 
that would allow local customization?

Reply:
@jedwards4b Hmm, could you elaborate?
From my understanding of your suggestion, I would prefer to only have to change a keyword value than having a user config file, but it is just my feeling.

Reply:
As for the comment from @mrocklin above, for me this could be put into the JobQueueCluster class, for choosing default processes and/or threads and memory. Maybe we ought to set those to None by default?
Commit incoming for demonstrating the automatic resource_spec assignment.

Reply:
If we cannot reach a common view about this, I propose to continue discussing it in another issue, maybe #12 or a new one. Then for this pull request, I can put resource_spec to None by default, remove the solution I proposed with my last commit if you don't like it, and eventually merge all this?

Reply:
These values are different from one machine to another, but typically don't need to change from one run to another on the same machine, sure a cli should be provided but a config file would allow the user not to have to remember and add the same cli again and again.